type-check.ts

```diff
@@ -212,7 +212,26 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<n
         throw new TypeCheckError("Condition Expression Must be a bool");
       return {a: NONE, tag:stmt.tag, cond: tCond, body: tBody};
     case "pass":
+    case "break":
+    case "continue":
       return {a: NONE, tag: stmt.tag};
+    case "for":
+      var tIterator = tcExpr(env, locals, stmt.iterator)
```

Comment:
Oh interesting idea here, since the variable has to be declared earlier and have a type, you can get the type here. Note that even if you restrict it to be a string (which we probably want to do, and then maybe generalize to destructuring assignment later), you can use the environment to find its type. That would make a bit more sense to me in the long run.

Reply:
Week-8 implementation considers iterator as string and its type is computed using environment variables. Thanks for the comment.